Dockerfile

```diff
@@ -0,0 +1,42 @@
+FROM ubuntu
+
+ENV DEBIAN_FRONTEND=noninteractive
+RUN apt-get update && \
+    apt-get install -y \
+        libffi6 pkg-config libffi-dev pkg-config build-essential \
+        gcc git wget make libncurses-dev flex bison gperf autoconf \
+        python3 python3-dev python3-pip python python-dev python-pip \
+        make unrar-free autoconf automake libtool gcc g++ gperf \
+        flex bison texinfo gawk ncurses-dev libexpat-dev \
+        sed git unzip bash help2man wget bzip2 \
+        libtool-bin gcc-arm-none-eabi qemu-system-arm
```

Comment:
The `gcc-arm-none-eabi` in ubuntu repositories is usually quote out of date and uncontrolled.

The official arm compiler should be used instead to use the best / controlled version: https://developer.arm.com/tools-and-software/open-source-software/developer-tools/gnu-toolchain/gnu-rm/downloads

Reply:
Should we use centos or another rolling release distro that contains newer versions of these dependencies?